core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/hash/HashProvider.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.user.core.hash;
+
+import org.wso2.carbon.user.core.exceptions.HashProviderException;
+
+import java.util.Map;
+
+/**
+ * This is the service interface for the HashProvider.
+ */
+public interface HashProvider {
+
+    /**
+     * Initialize the params from default values.
```

Comment:
`Initialize the HashProvider with default configuration values`

Reply:
resolved